x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/components/phases/hot_phase/use_rollover_value_required_validation.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useState, useEffect } from 'react';
+import { get } from 'lodash';
+import { useFormData, useFormContext, ValidationError } from '../../../../../../shared_imports';
+import { ROLLOVER_FORM_PATHS } from '../../../constants';
+import { ROLLOVER_VALUE_REQUIRED_VALIDATION_CODE } from '../../../form';
+
+const rolloverFieldPaths = Object.values(ROLLOVER_FORM_PATHS);
+
+export const useRolloverValueRequiredValidation = (): boolean => {
+  const [isValid, setIsValid] = useState(false);
+  const [formData] = useFormData({ watch: rolloverFieldPaths });
+  const form = useFormContext();
+
+  useEffect(() => {
+    // We check just the ROLLOVER_FORM_PATHS.maxPrimaryShardSize field because if
+    // it has the ROLLOVER_VALUE_REQUIRED_VALIDATION_CODE error, all the other rollover
+    // fields should too.
+    const rolloverFieldErrors: ValidationError[] =
+      get(form.getFields(), ROLLOVER_FORM_PATHS.maxPrimaryShardSize)?.errors ?? [];
+
+    setIsValid(
+      rolloverFieldErrors.some(
+        (validation) => validation.code === ROLLOVER_VALUE_REQUIRED_VALIDATION_CODE
+      )
+    );
+  }, [form, formData]);
```

Comment:
nit: to be slightly more precise and avoid triggering the effect whenever the form state changes you could just provide the `getFields` handler in the dependency and not the complete `form` object.